client/src/features/profile/setting/ProfileSetting.tsx

```diff
@@ -5,190 +5,305 @@ import SenpaiProfileSettingItems from './SenpaiProfileSettingItems';
+					}
+				);
+			}
+
+			await dispatch(getProfile());
+			await dispatch(getUsers());
+
+			setHasOverlay(false);
+
+			if (isSenpai) {
+				navigate('/profile/senpai');
+			} else {
+				navigate('/profile/kohai');
+			}
+		} catch (err: any) {
+			if (err.response) {
+				console.error(err.response.data);
+				if (err.response.data.message === 'Please enter a valid email') {
+					setErrorMessage(err.response.data.message);
+				} else {
+					setErrorMessage('Something went wrong...Please try again');
+				}
+			} else {
+				setErrorMessage('Something went wrong...Please try again');
+				console.error(err);
+			}
+			setHasOverlay(false);
+		}
+	};
+
+	const bgColor = () => {
+		return isSenpai ? 'bg-primary_bg_color' : 'bg-secondary_bg_color';
+	};
+
+	const profileName = () => {
+		if (isSenpai) {
+			return myProfile.name === ''
+				? `Senpai#${myProfile.senpaiProfile.id.slice(0, 5)}`
+				: myProfile.name;
+		} else {
+			return myProfile.name === ''
+				? `Kohai#${myProfile.kohaiProfile.id.slice(0, 5)}`
+				: myProfile.name;
+		}
+	};
+
+	return (
+		// TODO: adjust margin-top based on the height of the header
+		<>
+			{hasOverlay ? <LoadingScreen text="Uploading" /> : null}
+			<div
+				className={`${bgColor()} w-screen min-h-screen pt-mobileHeaderHeight lg:pt-laptopHeaderHeight`}
+			>
+				<div className="container max-w-xl mx-auto py-paddingAroundtheContent px-6 sm:px-8 flex flex-col gap-y-6">
+					<section className="flex gap-x-8 items-center">
+						<img
+							src={fileUrl ? fileUrl : myProfile.profileImage}
+							alt="profile"
+							className="object-cover w-28 h-28 md:w-40 md:h-40 lg:w-48 lg:h-48 rounded-full"
+						/>
+						<div>
+							<h1 className="text-xl font-bold mb-2">{profileName()}</h1>
+							<ToggleButton
+								isSenpai={isSenpai}
+								inputs={inputs}
+								setInputs={setInputs}
+							/>
+						</div>
+					</section>
+					<p className="text-red-500">*required</p>
+					<form className="flex flex-col gap-y-6">
+						<article>
+							<label htmlFor="name" className="font-bold nline-block">
+								*Name
+							</label>
+							<p className="text-sm">Please use your real name.</p>
+							<input
+								type="text"
+								id="name"
+								name="name"
+								value={inputs.name}
+								onChange={handleOnChange}
+								className="border border-gray-300 rounded focus:outline-none focus:border-primary_title_color"
+							/>
+						</article>
+						<article>
+							<label htmlFor="image" className="font-bold inline-block">
+								Profile Picture
+							</label>
+							<br />
+							<input
+								type="file"
+								id="image"
+								name="image"
+								accept="image/*"
+								onChange={handleOnChange}
+							/>
+							<div className="text-sm">
+								<p className="leading-snug">
+									The pre-set picture is generated by{' '}
+									<a
+										href="https://getavataaars.com/"
+										target="_blank"
+										rel="noreferrer"
```

Comment:
I just have one question about the `rel` attribute but you think we don't really need `noopener` anymore since some browsers don't support it and most of them support `noreferrer`? I was adding both like `rel="noopener noreferrer"`